src/datasets/arrow_dataset.py

```diff
@@ -1858,11 +1859,13 @@ def prepare_for_task(self, task: Union[str, TaskTemplate]) -> "Dataset":
             if not compatible_templates:
                 raise ValueError(f"Task {task} is not compatible with this dataset! Available tasks: {tasks}")
 
-            if len(compatible_templates) > 1:
+            if len(compatible_templates) > 1 and id is None:
```

Comment:
We could also return the id=0 by default, but no strong opinion on this

Reply:
My reasoning for `id=None` was to make selecting `id` more explicit, but this is also OK (`id=0` selects the most common task for a dataset).